lightning/src/ln/functional_tests.rs

```diff
@@ -6972,6 +6972,31 @@ fn test_set_outpoints_partial_claiming() {
 	}
 }
 
+#[test]
+fn test_counterparty_raa_skip_no_crash() {
+	// Previously, if our counterparty sent two RAAs in a row without us having provided a
+	// commitment transaction, we would have happily carried on and provided them the next
+	// commitment transaction based on one RAA forward. This would probably eventually have led to
+	// channel closure, but it would not have resulted in funds loss. Still, our
+	// EnforcingChannelKeys would have paniced as it doesn't like jumps into the future. Here, we
+	// check simply that the channel is closed in response to such an RAA, but don't check whether
+	// we decide to punish our counterparty for revoking their funds (as we don't currently
+	// implement that).
+	let node_cfgs = create_node_cfgs(2);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+	let chan = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::supported(), InitFeatures::supported());
+
+	let commitment_seed = nodes[0].node.channel_state.lock().unwrap().by_id.get_mut(&chan.2).unwrap().local_keys.commitment_seed().clone();
+	let next_per_commitment_point = PublicKey::from_secret_key(&Secp256k1::new(),
+		&SecretKey::from_slice(&chan_utils::build_commitment_secret(&commitment_seed, (1 << 48) - 3)).unwrap());
+	let per_commitment_secret = chan_utils::build_commitment_secret(&commitment_seed, (1 << 48) - 1);
```

Comment:
Could what's going on with `(1 << 48) - 3)` and `(1 << 48) - 3)` be made more explicit somehow? Not sure of the best way to do it. Could be as simple as well named variables... or better yet an abstraction that hides this math but makes it clear that the commitment number was advanced one too many times.

Reply:
A bit better? I copied the INITIAL_COMMITMENT_NUMBER const here to make the math clear.